storage/service_account/client_endpoint/client_endpoint_test.go

```diff
@@ -0,0 +1,39 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package clientendpoint
+
+import (
+	"bytes"
+	"os"
+	"strings"
+	"testing"
+)
+
+func TestSetClientEndpoint(t *testing.T) {
+	if os.Getenv("GOOGLE_APPLICATION_CREDENTIALS") == "" {
```

Comment:
This seems reasonable to me. Although using the other approach would be fine as well IMHO.

Reply:
Sounds good! Indeed no project ID is needed for this sample, though authentication is not avoided for `setClientEndpoint()`. So added a test skip to ensure the test passes under the right conditions.